zrml/swaps/src/migrations.rs

```diff
@@ -0,0 +1,110 @@
+pub mod _0_1_1 {
+    use crate::{BalanceOf, Config, Pallet, Pools};
+    use alloc::{collections::BTreeMap, vec::Vec};
+    use frame_support::{
+        dispatch::Weight,
+        traits::{Get, GetPalletVersion, PalletVersion},
+    };
+    use zeitgeist_primitives::types::{Asset, Pool, PoolStatus};
+
+    type PreviousPoolTy<T> = PreviousPool<BalanceOf<T>, <T as crate::Config>::MarketId>;
+
+    #[derive(
+        Clone,
+        Eq,
+        PartialEq,
+        parity_scale_codec::Decode,
+        parity_scale_codec::Encode,
+        sp_runtime::RuntimeDebug,
+    )]
+    struct PreviousPool<B, MI> {
+        assets: Vec<Asset<MI>>,
+        swap_fee: B,
+        total_weight: u128,
+        weights: BTreeMap<Asset<MI>, u128>,
+    }
+
+    pub fn migrate<T>() -> Weight
```

Comment:
Should this bump the pallet storage version?

Reply:
Some pallets do explicitly use storage versioning with an enum:

```rust
enum StorageVersion {
  V1,
  V2,
  ...
}
```

And others simply use the crate metadata version as a way to track upgrades, which is the chosen approach simply because it is a little easier, the default choice since https://github.com/paritytech/substrate/pull/7208 and we are temporarily migration a testnet

What I couldn't find with version-based migrations is the exposure of public API/Storage versions. In any case, I can use an `enum` to track storage versions if desirable.

Reply:
I see that the current variant is convenient in that it does not require to add a new storage value, but I think this will make updates more complicated overall. Please correct me if I am wrong, but as I see it now we have to update the pallet version for the next runtime upgrade, even if no changes to the pallet were made, because otherwise this condition:
https://github.com/zeitgeistpm/zeitgeist/blob/a144ed5a1c9cfcffdefd0f7e8beafd7797bd2ca5/zrml/swaps/src/migrations.rs#L35-L40
will be true again and the migration will be executed.

Reply:
Ah, I see what was my error in thought. The storage_version is a value inside storage that contains the pallet version before the runtime upgrade and it is updated after a runtime upgrade (in case the pallet version changed). This is smooth!